src/AddThreepid.ts

```diff
@@ -58,7 +58,7 @@ export type Binding = {
  * https://gist.github.com/jryans/839a09bf0c5a70e2f36ed990d50ed928
  */
 export default class AddThreepid {
-    private sessionId: string;
+    private sessionId?: string;
```

Comment:
It seems odd to me that we have a `?` here then lots of `!` later on. Couldn't we use the `!` here instead? 

Think I might be missing something here though

Reply:
The type can be undefined, the assertions later are because they'll be handled by the promise rejection

Reply:
Ok that makes sense - don't see any need to undo